test/functional/wallet_resendwallettransactions.py

```diff
@@ -39,6 +39,10 @@ def run_test(self):
         self.log.info("Create a new transaction and wait until it's broadcast")
         txid = int(node.sendtoaddress(node.getnewaddress(), 1), 16)
 
+        # Sleep for just over a second so the rebroadcast scheduler can get
```

Comment:
I don't get why you'd need to sleep before a wait_until statement. Isn't wait_until going to wait longer than 1.1 seconds for the condition to be true? It would be helpful to update this comment saying where the failure is and what happens if the sleep is not long enough.

Reply:
The `wait_until()` below can return in less than one second. If that happens then the rebroadcast timer won't have popped for the first time, and so `nNextResend` won't have been set. We then setmocktime, and when the rebroadcast timer does pop, `nNextResend` is still zero and `ResendWalletTransactions()` exits early.